src/test/java/core/basesyntax/RegistrationValidationTest.java

```diff
@@ -0,0 +1,148 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationValidationTest {
+    private static final StorageDao storageDao = new StorageDaoImpl();

```

Comment:
Consider using an instance field for 'storageDao' instead of a static field if it does not need to be shared among all instances. Static fields are shared across all instances of a class, which might not be the intended behavior for a DAO object.